donut/auth_utils.py

```diff
@@ -359,9 +368,31 @@ def is_caltech_user():
         re.search(ip_match, flask.request.remote_addr) is not None
 
 
-def is_admin():
+def is_admin(user_id=None):
     """
     Returns whether the current user is an admin
     """
-    username = flask.session.get('username')
+    if user_id:
+        username = get_username(user_id)
+    else:
+        username = flask.session.get('username')
     return username and check_permission(username, None)
+
+
+def get_admin_pos(user_id):
+    """
+    Returns position id and name of the admin position.
+    """
+    if not is_admin(user_id):
+        return []
```

Comment:
Is this check needed? I don't think the DB query below will return any rows if the user is not an admin. DB queries are generally slow, so it would be good to avoid them if possible.